src/pages-and-resources/teams/Settings.jsx

```diff
@@ -0,0 +1,150 @@
+    maxTeamSize: TeamSizes.DEFAULT,
+    id: null,
+  };
+
+  const handleSettingsSave = (values) => {
+    // For newly-added teams, fill in an id.
+    const topics = values.topics.map(topic => ({
+      id: topic.id || uuid(),
+      name: topic.name,
+      type: topic.type,
+      description: topic.description,
+      max_team_size: topic.maxTeamSize,
+    }));
+    saveSettings({ topics, max_team_size: values.maxTeamSize });
+  };
+
+  return (
+    <AppSettingsModal
+      appId="teams"
+      title={intl.formatMessage(messages.heading)}
+      enableAppHelp={intl.formatMessage(messages.enableTeamsHelp)}
+      enableAppLabel={intl.formatMessage(messages.enableTeamsLabel)}
+      learnMoreText={intl.formatMessage(messages.enableTeamsLink)}
+      onClose={onClose}
+      initialValues={{ ...teamsConfiguration }}
+      validationSchema={{
+        maxTeamSize: Yup.number()
+          .required(intl.formatMessage(messages.maxTeamSizeEmpty))
+          .min(TeamSizes.MIN)
+          .max(
+            TeamSizes.MAX,
+            intl.formatMessage(messages.maxTeamSizeTooHigh, {
+              max: TeamSizes.MAX,
+            }),
+          ),
+        topics: Yup.array().of(
+          Yup.object({
+            id: Yup.string(),
+            name: Yup.string()
+              .required(intl.formatMessage(messages.teamSetFormNameEmpty))
+              .trim(),
+            type: Yup.string().oneOf(Object.values(TeamSetTypes)),
+            description: Yup.string()
+              .required(intl.formatMessage(messages.teamSetFormDescriptionError))
+              .trim(),
+            maxTeamSize: Yup.number()
+              .min(TeamSizes.MIN)
+              .max(
+                TeamSizes.MAX,
+                intl.formatMessage(messages.maxTeamSizeTooHigh, {
+                  max: TeamSizes.MAX,
+                }),
+              )
+              .default(null),
+          }),
+        )
+          .default([])
+          .uniqueProperty('name', intl.formatMessage(messages.teamSetFormNameExists)),
+      }}
+      onSettingsSave={handleSettingsSave}
+      configureBeforeEnable
+    >
+      {
+        ({
+          handleChange, handleBlur, values, errors, touched,
+        }) => (
+          <>
+            <div>
+              <h4 className="mb-3">{intl.formatMessage(messages.teamSize)}</h4>
+              <Form.Group>
+                <Form.Control
+                  type="number"
+                  name="maxTeamSize"
+                  floatingLabel={intl.formatMessage(messages.maxTeamSize)}
+                  onChange={handleChange}
+                  onBlur={handleBlur}
+                  value={values.maxTeamSize}
+                />
+                {touched.maxTeamSize && errors.maxTeamSize && (
+                  <Form.Control.Feedback type="invalid" hasIcon={false}>
+                    {errors.maxTeamSize}
+                  </Form.Control.Feedback>
+                )}
+              </Form.Group>
+            </div>
```

Comment:
Fields and border spacing are not matching with figma. Helper texts are missing and do not match with figma.  Error message and helper text transition effect is not working also.

Current design:
<img width="750" alt="Screenshot 2021-09-10 at 3 02 27 PM" src="https://user-images.githubusercontent.com/79941147/132853600-37f3484d-8fb9-4380-80a8-307d08109527.png">
<img width="795" alt="Screenshot 2021-09-10 at 2 53 25 PM" src="https://user-images.githubusercontent.com/79941147/132853697-6a573a4c-f89b-420f-b02c-1a099f4552da.png">


Figma design:
<img width="428" alt="Screenshot 2021-09-10 at 5 31 54 PM" src="https://user-images.githubusercontent.com/79941147/132853653-6acfb06a-efbe-4e9d-80a3-55c012646d3d.png">
<img width="472" alt="Screenshot 2021-09-10 at 5 32 39 PM" src="https://user-images.githubusercontent.com/79941147/132853734-1ebd5620-2502-4d33-8079-68dbec124d7c.png">

Reply:
You do not need to make changes in paragon. You can use the bootstrap class for customization which is already using in discussion topics. Please have a look at the discussion section. let me know if you need any assistance.

Reply:
I'm not saying that making changes to paragon is a technical requirement, or that I don't know how to apply css classes. What I am considering is if the intent is to override the defaults from Pargaon (which would mean a less consistent UI across different MFEs and page) or to keep the defaults and get a more consistent appearance even if it doesn't match pixel-perfect.

Reply:
BTW. I have updated the PR to add the transitions, and updated the text. I'm not sure what you mean by border spacing and which helper text is missing.

Reply:
Just wanted to share the context that pages & resources apps view did generate a bunch of tickets for bringing that closer to mock-ups. 

I think we can leave this conversation for later on during the UI/UX review.